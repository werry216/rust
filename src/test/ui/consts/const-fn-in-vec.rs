fn main() {
    // should hint to create an inline const block
    // as all tests are on "nightly"
    let strings: [String; 5] = [String::new(); 5];
    //~^ ERROR the trait bound `String: Copy` is not satisfied
    println!("{:?}", strings);
}
