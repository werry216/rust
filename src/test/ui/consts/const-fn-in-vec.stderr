error[E0277]: the trait bound `String: Copy` is not satisfied
  --> $DIR/const-fn-in-vec.rs:4:32
   |
LL |     let strings: [String; 5] = [String::new(); 5];
   |                                ^^^^^^^^^^^^^^^^^^ the trait `Copy` is not implemented for `String`
   |
   = note: the `Copy` trait is required because the repeated element will be copied
   = help: create an inline `const` block, see PR #2920 <https://github.com/rust-lang/rfcs/pull/2920> for more information

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
